### ISSUE TITLE
Angular language service doesn't see components from library built with Ivy.

### ISSUE THREAD

**Issue by HackerMand** [angular/angular#34247]
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/language-service

### Description

Angular language service doesn't see components from library built with Ivy.
Application is built correctly.

## Error
<pre><code>
'lib-component' is not a known element:
1. If 'lib-component' is an Angular component, then verify that it is part of this module.
2. If 'lib-component' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.ng(0)
</code></pre>

## Environment

<pre><code>
Editor: VisualStudio Code
Editor extension: angular.ng-template v0.900.0

Angular CLI: 9.0.0-rc.5
Node: 10.17.0
OS: darwin x64

Angular: 9.0.0-rc.5
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.900.0-rc.5
@angular-devkit/build-angular      0.900.0-rc.5
@angular-devkit/build-ng-packagr   0.900.0-rc.5
@angular-devkit/build-optimizer    0.900.0-rc.5
@angular-devkit/build-webpack      0.900.0-rc.5
@angular-devkit/core               9.0.0-rc.5
@angular-devkit/schematics         9.0.0-rc.5
@angular/cdk                       9.0.0-rc.4
@ngtools/webpack                   9.0.0-rc.5
@schematics/angular                9.0.0-rc.5
@schematics/update                 0.900.0-rc.5
ng-packagr                         9.0.0-rc.3
rxjs                               6.5.3
typescript                         3.6.4
webpack                            4.41.2
</code></pre>

**Comment by HackerMand** [comment]
Update. Ivy compiler doesn't create file `lib-name.metadata.json` in `project_root/dist/lib-name`.
Could this be the problem?

**Comment by julien-von-oetinger** [comment]
+1 I have the same issue

**Comment by kyliau** [comment]
Post version 9, we are going to switch the language service internals to use the Ivy compiler. Until then, the language service would need to rely on `metadata.json` to retrieve additional information about a directive.

**Comment by jiverson** [comment]
Yes this is the issue. To resolve for the time being, you will need to build the prod version of your library which has `"enableIvy": false` in tsconfig. This then generates the metadata file needed in the dist folder. 

ie `ng build lib-component --prod`

But you also need to make sure to `Restart Angular Language service` after each build in vs-code. 

All this information is not easily found between the vs-code plugin and angular itself. Just a heads up for anyone else experiencing this same issue.
